packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -0,0 +1,376 @@
+    for (let rule of authRules) {
+      let operations: ModelOperation[] = rule.operations || MODEL_OPERATIONS;
+      if (rule.groups) {
+        rule.groups.forEach(group => {
+          let roleName = `${rule.provider}:staticGroup:${group}`;
+          if (!(roleName in this.roleMap)) {
+            this.roleMap.set(roleName, {
+              provider: rule.provider,
+              claim: rule.groupClaim || DEFAULT_GROUP_CLAIM,
+              value: group,
+            });
+          }
+          acm.setRole({ role: roleName, resource: field, operations });
+        });
+      } else {
+        var roleName: string;
+        let roleDefinition: any;
+        switch (rule.provider) {
+          case 'apiKey':
+            roleName = 'apiKey:public';
+            roleDefinition = { provider: rule.provider };
+            break;
+          case 'iam':
+            roleName = `iam:${rule.allow}`;
+            roleDefinition = {
+              provider: rule.provider,
+              claim: rule.allow === 'private' ? 'authenticated' : 'unauthenticated',
+            };
+            break;
+          case 'oidc':
+          case 'userPools':
+            if (rule.allow === 'groups') {
+              let groupsField = rule.groupsField || DEFAULT_GROUPS_FIELD;
+              roleName = `${rule.provider}:dynamicGroup:${groupsField}`;
+              roleDefinition = {
+                provider: rule.provider,
+                claim: rule.groupClaim || DEFAULT_GROUP_CLAIM,
+                value: groupsField,
+              };
+            }
+            if (rule.allow === 'owner') {
+              let ownerField = rule.ownerField || DEFAULT_OWNER_FIELD;
+              roleName = `${rule.provider}:owner:${ownerField}`;
+              roleDefinition = {
+                provider: rule.provider,
+                claim: rule.identityClaim || DEFAULT_IDENTITY_CLAIM,
+                value: ownerField,
+              };
+            } else {
+              throw new TransformerContractError(`Could not create a role from ${rule}`);
+            }
+            break;
+          default:
+            throw new TransformerContractError(`Could not create a role from ${rule}`);
+        }
+        acm.setRole({ role: roleName, resource: field, operations });
+        this.roleMap.set(roleName, roleDefinition);
+      }
+    }
+  }
+  private isAdminUIEnabled(): boolean {
+    return this.configuredAuthProviders.hasIAM && this.config.addAwsIamAuthInOutputSchema;
+  }
+  private extendAuthRulesForAdminUI(rules: Readonly<AuthRule[]>): AuthRule[] {
+    // Check for Amplify Admin
+    if (this.isAdminUIEnabled()) {
+      return [...rules, { allow: 'private', provider: 'iam', generateIAMPolicy: false }];
+    }
+    return [...rules];
+  }
+  private setAuthPolicyFlag(rules: AuthRule[]): void {
+    if (!rules || rules.length === 0 || this.generateIAMPolicyforAuthRole === true) {
```

Comment:
Since this is a private method, do we need the `!rules` check?

Reply:
Yea I don't think we need this anymore, since we are running auth config validation at the root level when creating a transformer class.